Content.Shared/Item/ItemToggle/SharedItemToggleSystem.cs

```diff
@@ -0,0 +1,313 @@
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ItemToggleComponent, UseInHandEvent>(OnUseInHand);
+        SubscribeLocalEvent<ItemToggleHotComponent, IsHotEvent>(OnIsHotEvent);
+        SubscribeLocalEvent<ItemToggleComponent, ItemUnwieldedEvent>(TurnOffonUnwielded);
+        SubscribeLocalEvent<ItemToggleComponent, ItemWieldedEvent>(TurnOnonWielded);
+        SubscribeLocalEvent<ItemToggleComponent, ItemToggleForceToggleEvent>(ForceToggle);
+    }
+    private void OnUseInHand(EntityUid uid, ItemToggleComponent itemToggle, UseInHandEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        args.Handled = true;
+
+        if (TryComp<WieldableComponent>(uid, out var wieldableComp))
+            return;
+
+        Toggle(uid, args.User, itemToggle);
+    }
+
+    public void Toggle(EntityUid uid, EntityUid? user = null, ItemToggleComponent? itemToggle = null)
+    {
+        if (!Resolve(uid, ref itemToggle))
+            return;
+
+        if (itemToggle.Activated)
+        {
+            TryDeactivate(uid, user, itemToggle: itemToggle);
+        }
+        else
+        {
+            TryActivate(uid, user, itemToggle: itemToggle);
+        }
+    }
+
+    public void ForceToggle(EntityUid uid, ItemToggleComponent itemToggle, ref ItemToggleForceToggleEvent args)
+    {
+        Toggle(uid, args.User, itemToggle);
+    }
+
+    public bool TryActivate(EntityUid uid, EntityUid? user = null, ItemToggleComponent? itemToggle = null)
+    {
+        if (!Resolve(uid, ref itemToggle))
+            return false;
+
+        if (itemToggle.Activated)
+            return true;
+
+        // The client cannot predict if the attempt to turn on fails or not since the battery and fuel systems are server side (for now). Potential future TODO
+        if (_netManager.IsServer)
+        {
+            var attempt = new ItemToggleActivateAttemptEvent(user);
+            RaiseLocalEvent(uid, ref attempt);
+
+            if (attempt.Cancelled)
+            {
+                //Play the failure to activate noise if there is any.
+                _audio.PlayPvs(itemToggle.SoundFailToActivate, uid);
+                return false;
+            }
+
+            // At this point the server knows that the activation went through successfully, so we play the sounds and make the changes.
+            _audio.PlayPvs(itemToggle.SoundActivate, uid);
+
+            // Starts the active sound (like humming).
+            if (TryComp(uid, out ItemToggleActiveSoundComponent? activeSound))
+            {
+                if (activeSound.ActiveSound != null && activeSound.PlayingStream == null)
+                {
+                    activeSound.PlayingStream = _audio.PlayPvs(activeSound.ActiveSound, uid, AudioParams.Default.WithLoop(true)).Value.Entity;
+                }
+            }
+
+            Activate(uid, itemToggle);
+            var ev = new ItemToggleActivatedEvent();
+            RaiseLocalEvent(uid, ref ev);
```

Comment:
This should just sub instead for the sound.

Reply:
Done.